CesiumRasterOverlays/src/RasterOverlayUtilities.cpp

```diff
@@ -339,4 +517,1138 @@ RasterOverlayUtilities::createRasterOverlayTextureCoordinates(
+      const T& v0 = accessor[vertex.first];
+      const T& v1 = accessor[vertex.second];
+      return glm::mix(v0, v1, vertex.t);
+    }
+  };
+
+  return std::visit(Operation{accessor}, vertex);
+}
+
+template <class T>
+T getVertexValue(
+    const AccessorView<T>& accessor,
+    const std::vector<CesiumGeometry::TriangleClipVertex>& complements,
+    const CesiumGeometry::TriangleClipVertex& vertex) {
+  struct Operation {
+    const AccessorView<T>& accessor;
+    const std::vector<CesiumGeometry::TriangleClipVertex>& complements;
+
+    T operator()(int vertexIndex) {
+      if (vertexIndex < 0) {
+        return getVertexValue(
+            accessor,
+            complements,
+            complements[static_cast<size_t>(~vertexIndex)]);
+      }
+
+      return accessor[vertexIndex];
+    }
+
+    T operator()(const CesiumGeometry::InterpolatedVertex& vertex) {
+      T v0{};
+      if (vertex.first < 0) {
+        v0 = getVertexValue(
+            accessor,
+            complements,
+            complements[static_cast<size_t>(~vertex.first)]);
+      } else {
+        v0 = accessor[vertex.first];
+      }
+
+      T v1{};
+      if (vertex.second < 0) {
+        v1 = getVertexValue(
+            accessor,
+            complements,
+            complements[static_cast<size_t>(~vertex.second)]);
+      } else {
+        v1 = accessor[vertex.second];
+      }
+
+      return glm::mix(v0, v1, vertex.t);
+    }
+  };
+
+  return std::visit(Operation{accessor, complements}, vertex);
+}
+
+template <class TIndex>
+bool upsamplePrimitiveForRasterOverlays(
+    const Model& parentModel,
+    Model& model,
+    Mesh& /*mesh*/,
+    MeshPrimitive& primitive,
+    CesiumGeometry::UpsampledQuadtreeNode childID,
+    bool hasInvertedVCoordinate,
+    const std::string_view& textureCoordinateAttributeBaseName,
+    int32_t textureCoordinateIndex) {
+  CESIUM_TRACE("upsamplePrimitiveForRasterOverlays");
+
+  // Add up the per-vertex size of all attributes and create buffers,
+  // bufferViews, and accessors
+  std::vector<FloatVertexAttribute> attributes;
+  attributes.reserve(primitive.attributes.size());
+
+  const size_t vertexBufferIndex = model.buffers.size();
+  model.buffers.emplace_back();
+
+  const size_t indexBufferIndex = model.buffers.size();
+  model.buffers.emplace_back();
+
+  const size_t vertexBufferViewIndex = model.bufferViews.size();
+  model.bufferViews.emplace_back();
+
+  const size_t indexBufferViewIndex = model.bufferViews.size();
+  model.bufferViews.emplace_back();
+
+  BufferView& vertexBufferView = model.bufferViews[vertexBufferViewIndex];
+  vertexBufferView.buffer = static_cast<int>(vertexBufferIndex);
+  vertexBufferView.target = BufferView::Target::ARRAY_BUFFER;
+
+  BufferView& indexBufferView = model.bufferViews[indexBufferViewIndex];
+  indexBufferView.buffer = static_cast<int>(indexBufferIndex);
+  indexBufferView.target = BufferView::Target::ELEMENT_ARRAY_BUFFER;
```

Comment:
This looks like the fix mentioned in PR description.
(marking for easy lookup later)